### DISCUSSION TITLE
set the proxy concurrently.

### DISCUSSION THREAD

**Discussion by niudaii** [projectdiscovery/nuclei#5316] [Ideas]
Maybe put types.ProxySocksURL and ProxyURL into the member variables of Options, instead of as global variables, because I want to set the proxy concurrently.

**Comment by GeorginaReeder** [comment]
Thanks for this suggestion @niudaii , we'll look into it!

**Reply by niudaii** [reply]
> Thanks for this suggestion @niudaii , we'll look into it!

Thanks!
Please also consider set **protocolstate.Dialer** into the member variables of Options, instead of as global variables too.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
